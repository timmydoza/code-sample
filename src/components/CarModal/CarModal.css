.modalOverlay {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: rgba(0, 0, 0, 0.8);
  display: none;
}

.displayModal {
  display: block;
}

.modal {
  position: absolute;
  width: 600px;
  top: 50%;
  right: 50%;
  transform: translate(50%, -50%);
  background: white;
  padding: 2em;
}

.modal ul {
  float: left;
}

.modal img {
  width: 50%;
  float: right;
}

.exit {
  position: absolute;
  top: 0.7em;
  left: 1em;
  font-size: 1.1em;
  border: 1px solid;
  border-radius: 100%;
  width: 1.3em;
  height: 1.3em;
  text-align: center;
  cursor: pointer;
}