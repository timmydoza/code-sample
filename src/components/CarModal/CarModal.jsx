import React from 'react';
import styles from './CarModal.css';
import classnames from 'classnames/bind';

const cx = classnames.bind(styles);

const CarModal = (props) => {

  const overlayClasses = cx(
    'modalOverlay',
    {
      'displayModal': props.selectedCarKey !== null
    }
  );

  return (
    <div className={overlayClasses} onClick={() => props.selectCar(null) }>
      <div className={styles.modal} onClick={(e) => e.stopPropagation() }>
        <span className={styles.exit} onClick={() => props.selectCar(null) }>x</span>
        <ul>
          <li>Year: {props.year}</li>
          <li>Make: {props.make}</li>
          <li>Model: {props.model}</li>
          <li>Mileage: {props.mileage}</li>
        </ul>
        <img src={props.image_url} />
      </div>
    </div>
  )
}

export default CarModal;