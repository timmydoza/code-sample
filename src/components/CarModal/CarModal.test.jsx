import React from 'react';
import renderer from 'react-test-renderer';
import CarModal from './CarModal';

const testCar = {
  make: 'TestMake',
  model: 'TestModel',
  year: 1999,
  mileage: 1234,
  image_url: 'image.jpg'
};

const nullVal = null; //Just using {null} below was messing up syntax highlighting.

describe('the CarModal component', () => {

  it('renders correctly', () => {
    const carModal = renderer
      .create(<CarModal {...testCar} selectCar={jest.fn()} selectedCarKey="1" />)
      .toJSON()
    expect(carModal).toMatchSnapshot();
  });

  it('should not have display class if selectedCarKey is null', () => {
    const carModal = renderer
      .create(<CarModal {...testCar} selectCar={jest.fn()} selectedCarKey={null} />).toJSON();
    expect(carModal.props.className).toBe('modalOverlay'); //theres no display class
  });

  it('should set the selectedCarKey to null when the user closes the modal', () => {
    const selectCar = jest.fn();
    const carModal = renderer
      .create(<CarModal {...testCar} selectCar={selectCar} selectedCarKey="1" />).toJSON();
    carModal.props.onClick();
    carModal.children[0].children[0].props.onClick(); //click both buttons
    expect(selectCar.mock.calls).toEqual([[null],[null]]);
  });

  it('clicks on the modal itself should not propagate', () => {
    const stopPropagation = jest.fn();
    const carModal = renderer
      .create(<CarModal {...testCar} selectCar={jest.fn()} selectedCarKey="1" />).toJSON();
    carModal.children[0].props.onClick({
      stopPropagation
    });
    expect(stopPropagation.mock.calls.length).toBe(1);
  });

});