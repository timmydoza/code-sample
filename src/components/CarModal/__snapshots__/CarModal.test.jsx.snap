// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the CarModal component renders correctly 1`] = `
<div
  className="modalOverlay displayModal"
  onClick={[Function]}
>
  <div
    className="modal"
    onClick={[Function]}
  >
    <span
      className="exit"
      onClick={[Function]}
    >
      X
    </span>
    <ul>
      <li>
        Year: 
        1999
      </li>
      <li>
        Make: 
        TestMake
      </li>
      <li>
        Model: 
        TestModel
      </li>
      <li>
        Mileage: 
        1234
      </li>
    </ul>
    <img
      src="image.jpg"
    />
  </div>
</div>
`;
